File: languages/go/cpu-prime-count/prime_count.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func countPrimes(limit int) int {
    if limit < 2 {
        return 0
    }
    sieve := make([]bool, limit+1)
    for i := 2; i <= limit; i++ {
        sieve[i] = true
    }
    for p := 2; p*p <= limit; p++ {
        if sieve[p] {
            for multiple := p * p; multiple <= limit; multiple += p {
                sieve[multiple] = false
            }
        }
    }
    count := 0
    for i := 2; i <= limit; i++ {
        if sieve[i] {
            count++
        }
    }
    return count
}

func main() {
    upper := 100000
    if len(os.Args) > 1 {
        if parsed, err := strconv.Atoi(os.Args[1]); err == nil {
            upper = parsed
        }
    }
    fmt.Println(countPrimes(upper))
}

File: languages/python/cpu-prime-count/prime_count.py
#!/usr/bin/env python3
import sys

def count_primes(limit: int) -> int:
    if limit < 2:
        return 0
    sieve = bytearray(b"\x01") * (limit + 1)
    sieve[0:2] = b"\x00\x00"
    p = 2
    while p * p <= limit:
        if sieve[p]:
            start = p * p
            step = p
            sieve[start:limit + 1:step] = b"\x00" * (((limit - start) // step) + 1)
        p += 1
    return sum(sieve)

if __name__ == "__main__":
    upper = int(sys.argv[1]) if len(sys.argv) > 1 else 100000
    print(count_primes(upper))
